core/src/main/scala/org/bykn/bosatsu/rankn/Infer.scala

```diff
@@ -509,6 +521,16 @@ object Infer {
             _ <- unify(tau, Type.Fun(argT, resT), fnRegion, evidenceRegion)
           } yield (argT, resT)
       }
+    def unifyFnForAllArg(fnType: Type, vars: NonEmptyList[Type.Var.Bound], fnRegion: Region, evidenceRegion: Region): Infer[(Type, Type)] =
+      fnType match {
+        case Type.Fun(Type.ForAll(vars2, arg), res) => pure((arg, res))
```

Comment:
I should probably look at the `vars2` and compare to `vars`

But as I say, i don't think this is the right approach. Just a step